lib/Serialization/DeserializeSIL.cpp

```diff
@@ -3459,18 +3459,17 @@ SILDeserializer::readDifferentiabilityWitness(DeclID DId) {
       ArrayRef<unsigned>(parameterAndResultIndices)
           .take_back(numResultIndices));
 
-  if (isDeclaration) {
-    auto *diffWitness = SILDifferentiabilityWitness::createDeclaration(
+  AutoDiffConfig config(parameterIndices, resultIndices, derivativeGenSig);
+  auto *diffWitness =
+      SILMod.lookUpDifferentiabilityWitness({originalName, config});
+  if (!diffWitness)
+    diffWitness = SILDifferentiabilityWitness::createDeclaration(
         SILMod, *linkage, original, parameterIndices, resultIndices,
         derivativeGenSig);
-    diffWitnessOrOffset.set(diffWitness, /*isFullyDeserialized*/ true);
-    return diffWitness;
-  }
-
-  auto *diffWitness = SILDifferentiabilityWitness::createDefinition(
-      SILMod, *linkage, original, parameterIndices, resultIndices,
-      derivativeGenSig, jvp, vjp, isSerialized);
-  diffWitnessOrOffset.set(diffWitness, /*isFullyDeserialized*/ true);
+  if (diffWitness->isDeclaration() && !isDeclaration)
```

Comment:
Could you please explain when this condition is true and `convertToDefinition` is called?
It doesn't seem wholly obvious, perhaps an explanatory comment would be good.

Reply:
done